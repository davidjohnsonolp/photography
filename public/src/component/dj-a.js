import buildShadowRoot from '../lib/build-shadow-root.js';

class DJA extends HTMLElement {
  constructor() {
    super();
    const html = `
      <style>
        :host {
        }
        a {
          color: royalblue;
          text-decoration: none;
        }
        a:hover {
          color: salmon;
          -webkit-transition: .2s linear 0s;
          -moz-transition: .2s linear 0s;
          -o-transition: .2s linear 0s;
          transition: .2s linear 0s;
        }
      </style>
      <a>
        <slot></slot>
      </a>
    `;
    buildShadowRoot(html, this);
    this.elems = {
      elem: this.shadowRoot.querySelector('selector')
    };
  }

  static get observedAttributes() {
    return ['href'];
  }

  attributeChangedCallback(attrName, oldVal, newVal) {
    switch(attrName) {
    case 'href':
      this.shadowRoot.querySelector('a').setAttribute('href',newVal);
      break;
    default:
      break;
    }
  }
}

customElements.define('dj-a', DJA);
export default DJA;
