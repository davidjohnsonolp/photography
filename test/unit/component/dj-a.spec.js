import '../../../public/src/component/dj-a.js';

let customAnchor;

describe('DJA Custom Anchor', () => {
  beforeEach(() => {
    customAnchor = window.document.createElement('dj-a'); 
    document.body.appendChild(customAnchor);
  });

  afterEach(() => {
    document.body.getElementsByTagName('dj-a')[0].remove();
  });

  it('renders correctly', () => {
    expect({ html: customAnchor.shadowRoot.innerHTML }).toMatchSnapshot();
  });

  it('renders correctly with href', () => {
    customAnchor.setAttribute('href', 'http://www.test.com/');

    expect({ html: customAnchor.shadowRoot.innerHTML }).toMatchSnapshot();
  });
});