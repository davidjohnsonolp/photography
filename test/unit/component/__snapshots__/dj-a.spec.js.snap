// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DJA Custom Anchor renders correctly 1`] = `
Object {
  "html": "
      <style>
        :host {
        }
        a {
          color: royalblue;
          text-decoration: none;
        }
        a:hover {
          color: salmon;
          -webkit-transition: .2s linear 0s;
          -moz-transition: .2s linear 0s;
          -o-transition: .2s linear 0s;
          transition: .2s linear 0s;
        }
      </style>
      <a>
        <slot></slot>
      </a>
    ",
}
`;

exports[`DJA Custom Anchor renders correctly with href 1`] = `
Object {
  "html": "
      <style>
        :host {
        }
        a {
          color: royalblue;
          text-decoration: none;
        }
        a:hover {
          color: salmon;
          -webkit-transition: .2s linear 0s;
          -moz-transition: .2s linear 0s;
          -o-transition: .2s linear 0s;
          transition: .2s linear 0s;
        }
      </style>
      <a href=\\"http://www.test.com/\\">
        <slot></slot>
      </a>
    ",
}
`;
